Add keycodes for pointing device scroll-mode hold

vnegnev opened the pull request:
PR adds keycodes that behave like the existing pointing device scroll-mode toggles, except they change the scroll-mode state both when pressed and when released, i.e. the state is changed while they are held down, and reverts when released. They do not interact with the EEPROM.

Also includes these keycodes, and the original scroll-mode toggles, in the set of keycodes that do not exit the automouse layer, so that they can be used with automouse.

Tested on a left-trackpoint Svalboard.

vnegnev committed:
Start Svalboard-specific firmware feature documentation

vnegnev committed:
Add buttons for scroll-mode hold

ilc commented on keyboards/svalboard/keymaps/keymap_support.c:
```diff
@@ -271,8 +277,14 @@ bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
-    if (record->event.pressed) {
+    if (record->event.pressed) { // key pressed
         switch (keycode) {
+            case SV_LEFT_SCROLL_HOLD:
```
Don't modify global_saved_values without calling write_eeprom_kb().

Make another global variable and, | the two values together.

vnegnev replied:
Done

ilc reviewed (changes requested):
Please make the needed changes.

vnegnev committed:
Write automouse allowed keycodes as text, minor text changes

vnegnev committed:
Add separate state to avoid temporarily modifying saved settings

vnegnev commented:
I've rebased off the docs branch, modified the code as advised to avoid using the saved values, and added v0.0.10 as a tag. Tested on a left-trackpoint Svalboard.

ilc commented on keyboards/svalboard/keymaps/keymap_support.h:
```diff
@@ -28,6 +28,8 @@ enum my_keycodes {
     SV_RIGHT_DPI_DEC,
     SV_LEFT_SCROLL_TOGGLE,
     SV_RIGHT_SCROLL_TOGGLE,
+    SV_LEFT_SCROLL_HOLD,
```
Key codes always go at the end, or they break all existing maps.

vnegnev replied:
Done, good catch. I've updated the order in both keymap_support.h and vial.mk.

ilc reviewed (changes requested):
Minor change, but needed,

vnegnev committed:
Put new keycodes at the end to avoid breaking keymaps; add comments

ilc reviewed (commented):
LGTM.
